Add `ContentUserRegistration.isCurrentlyRequired` field

B77Mills opened the pull request:
Add logic to only apply isRequired and access levels when the start and end date for requires Registration is met.

B77Mills committed:
update the userReg isRequired

add logic to only apply isRequired and access levels when the start and end date for requires Registration is met.

zarathustra323 commented on services/graphql-server/src/graphql/resolvers/platform/content.js:
```diff
@@ -535,6 +535,12 @@ module.exports = {
       const requiresRegistrationOptions = getAsObject(content, 'mutations.Website.requiresRegistrationOptions');
       const requiresAccessLevels = get(content, 'mutations.Website.requiresAccessLevels');
       const { startDate, endDate } = requiresRegistrationOptions;
+      if (requiresRegistration && (startDate || endDate)) {
```
I don't believe you want to change this resolver... if you do, the management interface will no longer function

zarathustra323 replied:
If you're trying to determine if registration is required for gating, that should be handled elsewhere

B77Mills committed:
add new isCurrentlyRequired to ContentUserRegistration

Add a new value for ContentUserRegistration to return a cacluated isCurrentlyRequired value based on if the start and end Date restrictions are met.

B77Mills committed:
fix isCurrentlyRequired

solocommand commented on services/graphql-server/src/graphql/resolvers/platform/content.js:
```diff
@@ -310,6 +310,14 @@ module.exports = {
+      const now = new Date();
+      if (root.startDate && root.startDate <= now) return false;
+      if (root.endDate && root.endDate >= now) return false;
+      return root.isRequired;
```
I think your logic is reversed here; you want to return true if there is a start date and it's in the past. You'd need to check the end date to be sure in that condition though. Something like this?
```suggestion
      if (root.startDate && root.startDate <= now) {
        if (root.endDate && root.endDate <= now) return false;
      }
      return true;
```

B77Mills replied:
@solocommand I did have the checks wrong, but I think they still are independent conditions.  I changed it to 

```
if (root.startDate && root.startDate > now) return false;
if (root.endDate && root.endDate < now) return false;
```

zarathustra323 commented on services/graphql-server/src/graphql/resolvers/platform/content.js:
```diff
@@ -310,6 +310,14 @@ module.exports = {
    *
    */
   ContentUserRegistration: {
+    isCurrentlyRequired: async (root) => {
```
Destructure `root` and update the ifs to use the vars directly...
```js
isCurrentlyRequired: async ({ isRequired, startDate, endDate }) => {
```

zarathustra323 commented on services/graphql-server/src/graphql/resolvers/platform/content.js:
```diff
@@ -530,11 +538,18 @@ module.exports = {
 
     userRegistration: (content) => {
       const requiresRegistration = get(content, 'mutations.Website.requiresRegistration');
-      if (!requiresRegistration) return { isRequired: false, siteIds: [], accessLevels: [] };
```
Theses changes can be discarded

B77Mills committed:
Destructure root obj

B77Mills committed:
revert this change
